frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -199,6 +215,34 @@ public void run(ProgressIndicator indicator) {
                 getMessageOrEmpty(t))));
       }
 
+      @SuppressWarnings("regexp") // needed to use forbidden "synchronized"
+      private void waitForLocalBranch() {
+        Supplier<@Nullable GitLocalBranch> findLocalBranch = () -> selectedVcsRepository.getBranches()
+            .findLocalBranch(slideInOptions.getName());
+
+        try {
+          //  6 attempts, usually 3 are enough
+          final int TIMEOUT = 2048;
+          long WAIT_DURATION = 64;
+          while (findLocalBranch.get() == null && WAIT_DURATION <= TIMEOUT) {
+            synchronized (this) {
```

Comment:
Hmm why specifically `synchronized` around `wait`? As in, what's specifically the shared resource you're trying to protect from concurrent access here?

Reply:
`wait()` must be called within a `synchronized` block.
Otherwise, we will get an `IllegalMonitorStateException`.

Ref:
https://stackoverflow.com/questions/2779484/why-must-wait-always-be-in-synchronized-block

Reply:
See my recent comments on `wait`